### PR TITLE
Add a dry-run feature flag to test incremental rollouts

### PR DESCRIPTION
**Pull request by samsymons**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/0/1205288845646357/f
iOS PR: https://github.com/duckduckgo/iOS/pull/1921
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1493
What kind of version bump will this require?: Major

**Optional**:

Tech Design URL:
CC: @SlayterDev 

**Description**:

This PR updates the phased rollout feature flag feature to match the Android JSON schema.

It also adds a new feature/subfeature pair, to test the feature in production later in the month.

**Steps to test this PR**:
1. See client-side PRs for per-platform testing steps

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/Configuration/ConfigurationFetching.swift [hunk 1]
```diff
@@ -135,7 +135,8 @@ public final class ConfigurationFetcher: ConfigurationFetching {
         let log = log
         let request = APIRequest(configuration: configuration, requirements: requirements, urlSession: urlSession, log: log)
         let (data, response) = try await request.fetch()
-        return (response.etag!, data)
+        return ("some-etag", data)
+        // return (response.etag!, data)
```
Comment (samsymons) [inline Sources/Configuration/ConfigurationFetching.swift:139]: TODO: Undo this change, this just makes it possible to test with a modified config from jsonblob.

File: Sources/Networking/APIRequest.swift [hunk 2]
```diff
@@ -100,7 +100,7 @@ public struct APIRequest {
         }
         
         if requirements.contains(.requireETagHeader), httpResponse.etag == nil {
-            throw APIRequest.Error.missingEtagInResponse
+            // throw APIRequest.Error.missingEtagInResponse
```
Comment (samsymons) [inline Sources/Networking/APIRequest.swift:103]: TODO: Undo this change, this just makes it possible to test with a modified config from jsonblob.

**Review by SlayterDev — APPROVED** [review]
LGTM! Don't forget about the todo items 😁

**Comment by samsymons** [comment]
This made it into `main` via https://github.com/duckduckgo/BrowserServicesKit/pull/465, so I'm closing this.
